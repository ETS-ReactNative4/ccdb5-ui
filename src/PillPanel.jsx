import React from 'react';
import { connect } from 'react-redux'
import { knownFilters } from './constants'
import Pill from './Pill';
import './PillPanel.less';

export const PillPanel = ({ filters, clearAll }) => {
  if( !filters || filters.length === 0 ) {
    return null
  }

  return (
    <section className="pill-panel">
      <h5 className="pill-label flex-fixed">Filters Applied:</h5>
      <ul className="layout-row">
        { filters.map(x => <Pill key={x.fieldName + x.value}
                                 fieldName={x.fieldName}
                                 value={x.value} />)
        }
        <li className="clear-all">
          <button className="a-btn a-btn__link body-copy" onClick={ clearAll }>
            Clear all filters
          </button>
        </li>
      </ul>
    </section>
  );
}

export const mapStateToProps = (state) => {
  const subState = state.query || {}
  const filters = knownFilters
    // Only use the known filters that are in the substate
    .filter(x => {
      return x in subState
    })
    // Create a flattened array of pill objects
    .reduce((accum, fieldName) => {
      const arr = subState[fieldName].map(
        value => { return {fieldName, value} }
      )
      return accum.concat(arr)
    }, [])

  return {
    filters: filters
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    clearAll: console.log('clearAll')
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PillPanel);
