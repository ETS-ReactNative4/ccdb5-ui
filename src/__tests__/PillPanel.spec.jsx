import React from 'react';
import { PillPanel } from '../PillPanel';
import renderer from 'react-test-renderer';

describe('component:PillPanel', () => {
  it('renders without crashing', () => {
    const filters = {
      alpha: 'Apples',
      beta: 'Bananas are great',
    };

    const target = renderer.create(
      <PillPanel filters={filters} />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
