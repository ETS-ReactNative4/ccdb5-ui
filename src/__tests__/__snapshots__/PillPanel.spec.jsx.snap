exports[`component:PillPanel does not render when there are no filters 1`] = `null`;

exports[`component:PillPanel renders without crashing 1`] = `
<section
  className="pill-panel">
  <h5
    className="pill-label flex-fixed">
    Filters Applied:
  </h5>
  <ul
    className="layout-row">
    <li
      className="pill flex-fixed">
      <span
        className="name">
        Apples
      </span>
      <button
        onClick={undefined}
        title="Remove Apples as a filter">
        <span
          className="cf-icon cf-icon-delete" />
      </button>
    </li>
    <li
      className="pill flex-fixed">
      <span
        className="name">
        Bananas are great
      </span>
      <button
        onClick={undefined}
        title="Remove Bananas are great as a filter">
        <span
          className="cf-icon cf-icon-delete" />
      </button>
    </li>
    <li
      className="clear-all">
      <button
        className="a-btn a-btn__link body-copy"
        onClick={undefined}>
        Clear all filters
      </button>
    </li>
  </ul>
</section>
`;
